src/index.js

```diff
@@ -0,0 +1,82 @@
+import { sha256, sha512 } from 'multiformats/hashes/sha2'
+import { murmur3128, murmur332 } from '@multiformats/murmur3'
+import { blake2b256 } from '@multiformats/blake2/blake2b'
+import { blake2s256 } from '@multiformats/blake2/blake2s'
+import {
+  sha3224,
+  sha3256,
+  sha3384,
+  sha3512,
+  shake128,
+  shake256,
+  keccak224,
+  keccak256,
+  keccak384,
+  keccak512
+// @ts-expect-error types not well published
+} from '@multiformats/sha3'
+import { equals } from 'uint8arrays'
+
+/**
+ * @typedef {object} Block
+ * @property {import('multiformats/cid').CID} cid
+ * @property {Uint8Array} bytes
+ */
+
+/**
+ * Validates IPLD block bytes.
+ * @param {Block} block
+ */
+export async function validateBlock (block) {
+  const hasher = getHasher(block.cid.multihash.code)
+  console.log('hasher', hasher, block.bytes)
+  const bytesHash = await hasher.digest(block.bytes)
+
+  console.log('has', bytesHash)
+
+  if (!equals(bytesHash.digest, block.cid.multihash.digest)) {
+    throw new Error(`block with cid ${block.cid.toString()} does not have valid bytes`)
+  }
+}
+
+/**
+ * @param {number} multihashCode
+ */
+function getHasher (multihashCode) {
+  switch (multihashCode) {
+    case sha256.code:
+      return sha256
+    case sha512.code:
+      return sha512
+    case murmur3128.code:
+      return murmur3128
+    case murmur332.code:
+      return murmur332
```

Comment:
it's really hard to spot if we get one of these wrong... I kinda love this style, where you convert a list of things to the map init format:
```
const hashMap = new Map([sha256, sha512, murmer3128, ...].map(hash => [hash.code, hash]))
```

e.g https://github.com/web3-storage/linkdex/pull/12/files#diff-c73abcf5e120d37e1deaae000d2b9fa15409ea2829918774125681f66cf76c2dR14